client/client_with_secret_test.go

```diff
@@ -215,3 +217,43 @@ func Test_GivenTokenSourceWithTokenThatWillNotExpireBetweenRequests_WhenTokenSto
 	mockedAuthService.AssertExpectations(t)
 	mockedClient.AssertExpectations(t)
 }
+
+func Test_GivenAServerThatRejectsHTTPCall_WhenAGetCallIsFired_ThenExpectTheClientToRenewAccessTokenOnce(t *testing.T) {
+	testCases := []struct {
+		expectedStatusCode int
+		expectedNoOfCalls  int
+	}{
+		{http.StatusUnauthorized, 2},
+		{http.StatusOK, 1},
+	}
+
+	for _, testCase := range testCases {
+		// Given
+		mockedAuthService := mocks.AuthService{}
+		mockedClient := mocks.Client{}
+
+		accessToken := "initial-access-token"
+
+		ts := startMockServer(t, &mockedAuthService, &mockedClient, accessToken, http.StatusOK, testCase.expectedStatusCode)
+		defer ts.Close()
+
+		mockedAuthService.
+			On("Token").Return().
+			Times(testCase.expectedNoOfCalls)
+		mockedClient.
+			On("Test", "initial-access-token").Return().
+			Times(testCase.expectedNoOfCalls)
+
+		// When
+		c := client.NewWithSecret("my-client-id", "my-secret",
+			client.WithBaseURL(ts.URL)).HTTPClient()
+
+		// Then
+		resp, err := c.Get(ts.URL)
```

Comment:
Just two small things
- shouldn't `resp, err := c.Get(ts.URL)` be in the `When`
- and`c := client.NewWithSecret...` in the `Given` part?